### ISSUE TITLE
Rails 4.2 compatibility

### ISSUE THREAD

**Issue by matfiz** [adzap/validates_timeliness#121]
I am successfully using this gem with Rails 4.1.4. However, every time I validate something in a model using the validators provided by this gem, in the logs I get the *DEPRECATION WARNING*:

DEPRECATION WARNING: The `Validator#setup` instance method is deprecated and will be removed on Rails 4.2. Do your setup in the constructor instead:

```ruby
class MyValidator < ActiveModel::Validator
  def initialize(options={})
    super
    options[:class].send :attr_accessor, :custom_attribute
  end
end
```

**Comment by CUnknown** [comment]
I would also like to see an update on this issue.

**Comment by guapolo** [comment]
Having the same issue as @kreintjes on an hstore/store_accesor validation

```
Exception: NoMethodError: undefined method `type' for "date":String
--
0: /gems/activerecord-4.2.0/lib/active_record/connection_adapters/column.rb:18:in `type'
1: /gems/jc-validates_timeliness-3.1.1/lib/validates_timeliness/orm/active_record.rb:14:in `timeliness_attribute_type'
2: /gems/jc-validates_timeliness-3.1.1/lib/validates_timeliness/orm/active_record.rb:33:in `timeliness_type_cast_code'
3: /gems/jc-validates_timeliness-3.1.1/lib/validates_timeliness/attribute_methods.rb:48:in `define_timeliness_write_method'
4: /gems/jc-validates_timeliness-3.1.1/lib/validates_timeliness/attribute_methods.rb:38:in `define_attribute_timeliness_methods'
```

I tried to find the bug, no luck.

**Comment by kernel-io** [comment]
https://github.com/kernel-io/validates_timeliness

fixed ( I think, its late at night here). Give it a whirl.

**Comment by kreintjes** [comment]
@kernel-io seems to work. Thanks!

**Comment by rhomeister** [comment]
@kernel-io Excellent work. @johncarney Can we merge this into  https://github.com/johncarney/validates_timeliness and release? This gem is currently broken for Rails 4.2 under the conditions that @guapolo and @kreintjes mentioned. I use this gem heavily in my apps, so a proper release would be very much appreciated!

**Comment by tibbon** [comment]
+1 for a merger

**Comment by pkmiec** [comment]
I've created https://github.com/appfolio/validates_timeliness which combines all the Rails 4.x patches, STI patch, and performance patches. It also appraises against rails 4.0, 4.1, 4.2 (dropped support for < 4.0) and ruby 2.0, 2.1, 2.2 (dropped support for < 2.0). I dropped support for mongo, and the extensions since I don't use them and they break in rails 4.2.

The gem is released as ae-validates_timeliness.

**Comment by dgilperez** [comment]
+1 for a merger, now leaning towards @pkmiec fork.

**Comment by adzap** [comment]
I have released v4 for Rails 4.0, 4.1 and 4.2. Please add new issues as you find them.

**Comment by rusllonrails** [comment]
:+1: :santa:
